Geometry.cs

```diff
@@ -144,14 +158,46 @@ class Drawing {
 
    public Bound2 Bound {
       get {
-         if (mBound.IsEmpty) mBound = new (Polys.Select (a => a.Bound));
+         if (mBound.IsEmpty) mBound = new (ConvexHull);
          return mBound;
       }
    }
    Bound2 mBound;
 
-   public Bound2 GetBound (Matrix2 xfm) 
-      => new Bound2 (Polys.SelectMany (a => a.Pts.Select (p => p * xfm)));
+   IReadOnlyList<Point2> ConvexHull { 
+      get {
+         if (!mConvexHull.Any ()) ComputeConvexHull (Polys.SelectMany (a => a.Pts));
+         return mConvexHull;
+      } 
+   }
+   List<Point2> mConvexHull = new ();
+
+   void ComputeConvexHull (IEnumerable<Point2> pts) {

```

Comment:
This function sets 'mConvexHull' as a side-effect. That is not obvious from the calling code, and it is not a good practice. Have the function return the list that you then store into mConvexHull. Alternatively, rename the function as UpdateConvexHull.

Reply:
Changed method name